### DISCUSSION TITLE
Airgapped environment DB update

### DISCUSSION THREAD

**Discussion by Trivynoobie** [aquasecurity/trivy#3207] [Q&A] [answered]
Hello Trivy community,

I work in an air gapped environment and I followed the instructions. I'm using version 0.34.0 and have tried this with a couple different versions after 0.28.1. After copying the trivy-offline.db.tgz to '/home/<user>/.cache/trivy/db' and 'tar -zxvf trivy-offline.db.tgz' it is giving me fatal errors. 

trivy image --skip-update --offline-scan localhost/<container name>

ERROR  The local DB has an old schema version which is not supported by the current version of Trivy CLI. DB needs to be updated.
FATAL   init error: DB error: database error: validate error: --skip-update cannot be specified with the old DB schema. Local DB: 1, Expected: 2

trivy image --skip-update --offline-scan --skip-policy-update localhost/<container name>

ERROR  The first run cannot skip download DB
FATAL  init error: DB error: database error: --skip-update cannot be specified on the first run

Thoughts? Thank you.

**Comment by afdesk** [comment, marked as answer]
@Trivynoobie it seems you got the old version of Trivy DB.
I need to use something like `oras` for downloading:
```sh
$ oras pull ghcr.io/aquasecurity/trivy-db:2
```

read more about Air-Gapped environment [here](https://aquasecurity.github.io/trivy/v0.34/docs/advanced/air-gap/)

**Reply by Trivynoobie** [reply]
Thank you for the response. Is there no website we can download from? Something like https://github.com/aquasecurity/trivy-db/releases/tag/v1-2022112212 ? I checked https://github.com/aquasecurity/trivy-db/pkgs/container/trivy-db/52518365?tag=2 but it doesn't seem to have a website download. I don't have a Linux machine that is connected to the internet but the machine running my containers is Linux.

**Reply by afdesk** [reply]
@Trivynoobie Now Trivy DB is an artifact on GHCR. so you need to download it 
but you can use Trivy for it:
```
$ trivy i --download-db-only --cache-dir ~/.cache/downloaded-trivy-db
```
`~/.cache/downloaded-trivy-db/db` folder will contain Trivy db files

**Reply by Trivynoobie** [reply]
It worked! Do I need to do anything with fanal.db? I noticed that was downloaded as well. It appears to be for "go" but doesn't affect scanning. Thank you.

**Reply by afdesk** [reply]
`fanal.db` is Trivy's cache. it doesn't need to copy
